apps/emqx/src/emqx_channel.erl

```diff
@@ -552,7 +552,7 @@ process_publish(Packet = ?PUBLISH_PACKET(QoS, Topic, PacketId), Channel) ->
                 msg => "cannot_publish_to_topic",
                 topic => Topic,
                 reason => emqx_reason_codes:name(Rc)
-            }),
+            }, #{topic => Topic}),
```

Comment:
Duplicated with line 553

Reply:
NO, the meta is for filter.
553 is for message display.

Reply:
Meta data is eventually merged into Data for logging